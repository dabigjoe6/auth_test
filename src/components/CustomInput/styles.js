const styles = {
  container: {
    marginBottom: 20,
  },
  label: {
    fontSize: 16,
    marginBottom: 10,
  },
  textInput: {
    paddingHorizontal: 10,
    borderRadius: 4,
  },
  error: {
    color: 'red',
  },
};

export default styles;
