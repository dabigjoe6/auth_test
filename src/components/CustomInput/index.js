/* eslint-disable react-native/no-inline-styles */
import React, {useState} from 'react';
import {Text, View} from 'react-native';
import {TextInput} from 'react-native-gesture-handler';

import styles from './styles';

const CustomInput = React.forwardRef(({label, error, ...otherProps}, ref) => {
  const [isFocussed, toggleFocus] = useState(false);
  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <TextInput
        {...otherProps}
        ref={ref}
        onFocus={() => toggleFocus(true)}
        onBlur={() => toggleFocus(false)}
        style={{
          ...styles.textInput,
          backgroundColor: isFocussed ? '#FFFFFF' : '#EAEAEB',
        }}
      />
      {error && <Text style={styles.error}>{error}</Text>}
    </View>
  );
});

export default CustomInput;
