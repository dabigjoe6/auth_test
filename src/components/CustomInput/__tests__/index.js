import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomInput from '..';
import {TextInput, Text} from 'react-native';

describe('Custom Input', () => {
  it('should render without issues', () => {
    const wrapper = shallow(<CustomInput label="Test" />);

    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render style onFocus', () => {
    const wrapper = shallow(<CustomInput label="Test" />);

    wrapper.find(TextInput).simulate('focus');
    wrapper.update();
    expect(wrapper.find(TextInput).prop('style')).toHaveProperty(
      'backgroundColor',
      '#FFFFFF',
    );
  });

  it('should render style onBlur', () => {
    const wrapper = shallow(<CustomInput label="Test" />);

    wrapper.find(TextInput).simulate('blur');
    wrapper.update();
    expect(wrapper.find(TextInput).prop('style')).toHaveProperty(
      'backgroundColor',
      '#EAEAEB',
    );
  });

  it('should render error', () => {
    const wrapper = shallow(<CustomInput label="Test" />);

    wrapper.setProps({error: 'Error test'});
    wrapper.update();
    expect(wrapper.find(Text).children().contains('Error test')).toBeTruthy();
    // expect(wrapper.find(Text).children).
  });
});
