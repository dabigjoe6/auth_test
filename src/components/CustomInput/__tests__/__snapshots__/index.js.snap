// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input should render without issues 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Test
  </Text>
  <Component
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#EAEAEB",
        "borderRadius": 4,
        "paddingHorizontal": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
